blockpoller/poller_test.go

```diff
@@ -157,7 +157,7 @@ func TestForkHandler_run(t *testing.T) {
 			f := New(blockFetcher, blockFinalizer)
 			f.forkDB = forkable.NewForkDB()
 
-			err := f.run(tt.startBlock)
+			err := f.run(tt.startBlock, 1)
```

Comment:
shouldn't we write a test for the optimistic poller too ? seems tricky enough to deserve a test